plugins/API/DataTable/MergeDataTables.php

```diff
@@ -52,4 +52,23 @@ public function mergeDataTables($table1, $table2)
         }
     }
 
+    private function makeNewDataTable(DataTable\DataTableInterface $subTable2)
+    {
+        if ($subTable2 instanceof DataTable\Map) {
```

Comment:
@diosmosis wondering if we would need to create a clone and keep any filters? Not sure though

Reply:
I guess we could. Might be an issue if filters hold references to the old table or to some other object that isn't relevant anymore (ie, like ones that calculate percent over a total or specific column value in another table or something like that). Actually, it's probably better to require this class be used before filters are run, since if metrics are formatted they won't be able to be added properly...

Reply:
It shouldn't hold a reference to the table but possibly to some other parameters.

Reply:
Some filters do hold references to other tables, eg, CalculateEvolutionFilter. It is conceivable that they could hold onto a lot of data.

Ideally it would be great to deprecate queued filters and instead move that logic to a method in the Report class, like, `formatApiResponse()`, similar to `configureView()` or whatever it's called.